build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,157 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags, Config } from "@oclif/core";
+import { Package, updatePackageJsonFile } from "@fluidframework/build-tools";
+import path from "node:path";
+import { PackageCommand } from "../../BasePackageCommand";
+// eslint-disable-next-line node/no-missing-import
+import type { PackageJson } from "type-fest";
+import { ExtractorConfig } from "@microsoft/api-extractor";
+import { CommandLogger } from "../../logging";
+
+interface PackageTypesList {
+	packagesUpdated: string[];
+	packagesNotUpdated: string[];
+}
+
+interface UpdateConfig {
+	pkg: Package;
+	releaseType: string;
+	json: PackageJson;
+	typesFolder: string;
+	includesAlphaBeta: boolean;
+}
+
+const knownReleaseTag = ["alpha", "beta", "public"] as const;
+
+type ReleaseTag = typeof knownReleaseTag[number];
+
+function isReleaseTag(str: string | undefined): str is ReleaseTag {
+	return str === undefined ? false : knownReleaseTag.includes(str as any);
+}
+
+export default class SetReleaseTagPublishingCommand extends PackageCommand<
+	typeof SetReleaseTagPublishingCommand
+> {
+	static readonly description =
+		"Updates the .d.ts types that are included in the `types` field in package.json. This command is intended to be used in CI only.";
+
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		types: Flags.custom<ReleaseTag>({
+			description: "Which .d.ts types to include in the published package.",
+			parse: async (input) => {
+				if (isReleaseTag(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+			required: true,
+		})(),
```

Comment:
```suggestion
		}),
```

I don't think you need to call this.